desc/magnetic_fields.py

```diff
@@ -1094,3 +1130,644 @@ def odefun(rpz, s):
     r = x[:, :, 0].T.reshape((len(phis), *rshape))
     z = x[:, :, 2].T.reshape((len(phis), *rshape))
     return r, z
+
+
+class CurrentPotentialField(_MagneticField, FourierRZToroidalSurface):
+    """Magnetic field due to a surface current potential on a toroidal surface.
+
+        surface current K is assumed given by
+         K = ∇ Φ x n
+        where:
+               n is the winding surface unit normal
+               Phi is the current potential function,
+               which is a function of theta and zeta
+        this function then uses biot-savart to find the
+        B field from this current density K on the surface
+
+    Parameters
+    ----------
+    potential : callable
+        function to compute the current potential. Should have a signature of
+        the form potential(theta,zeta,**params) -> ndarray.
+        theta,zeta are poloidal and toroidal angles on the surface
+    potential_dtheta: callable
```

Comment:
can you explain a bit why we can't use AD to get the derivatives? I thought something like we do for `ScalarPotentialField` would work?

Reply:
It does, but it gets trickier when we want to use the compute index as well, we need params to be something that can be passed in, but to AD with keyword arguments (at least from what I was seeing), it was easier to use `functools.partial` to remove them first before doing the AD, and that means that any times `params` changes, we need to re-create the derivatives

it is probably possible but also I was just having a harder time dealing with it, so I'd rather get this in without it then come back to adding AD later